mango-core/src/main/scala/org/bdgenomics/mango/models/AlignmentRecordMaterialization.scala

```diff
@@ -128,8 +176,21 @@ class AlignmentRecordMaterialization(s: SparkContext, d: SequenceDictionary, par
     seqRecord match {
       case Some(_) => {
         val (reg, ref) = refRDD.getPaddedReference(region)
+
         ks.map(k => {
-          val data = loadFromFile(region, k).map(r => (r._1, CalculatedAlignmentRecord(r._2, MismatchLayout(r._2, ref.get, reg))))
+          val alignmentData = loadFromFile(region, k)
+            .map(r => (ReferenceRegion(r), r))
```

Comment:
why the decision to put the map here? Is there actually any way we can avoid the ReferenceRegion construction? Just a thought.